optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/drools/testgen/reproducer/TestGenCorruptedScoreReproducer.java

```diff
@@ -66,10 +66,10 @@ public boolean isReproducible(TestGenKieSessionJournal journal) {
         } catch (RuntimeException e) {
             if (e.getMessage() != null && e.getMessage().startsWith("No fact handle for ")) {
                 // this is common when removing insert of a fact that is later updated - not interesting
-                logger.debug("    Can't remove insert: {}", e.toString());
+                logger.debug("    Can't remove insert: {}", (Object) e);
```

Comment:
This does appear to be a difference between backends.

The "simple" backend's implementation of debug(String, Object) passes an extra `null` parameter to MessageFormatter.format():  
https://github.com/qos-ch/slf4j/blob/1c13755b2520e511bc41b5c6fdc8bd6c58e529c0/slf4j-simple/src/main/java/org/slf4j/simple/SimpleLogger.java#L445

The "log4j12" backend does not:  
https://github.com/qos-ch/slf4j/blob/1c13755b2520e511bc41b5c6fdc8bd6c58e529c0/slf4j-log4j12/src/main/java/org/slf4j/log4j12/Log4jLoggerAdapter.java#L227

There is also a related bug report and open pull request: qos-ch/slf4j#166

When that PR is merged, it should be okay to cast to Object rather than call toString() because the exception will be "consumed" by the format string.